modules/rostests/apitests/advapi32/RegCreateKeyEx.c

```diff
@@ -0,0 +1,186 @@
+    ZeroMemory(&ea, sizeof(ea));
+    ea[0].grfAccessPermissions = KEY_READ;
+    ea[0].grfAccessMode = SET_ACCESS;
+    ea[0].grfInheritance= NO_INHERITANCE;
+    ea[0].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[0].Trustee.TrusteeType = TRUSTEE_IS_WELL_KNOWN_GROUP;
+    ea[0].Trustee.ptstrName  = pEveryoneSID;
+
+    // Create a SID for the BUILTIN\Administrators group.
+    bRes = AllocateAndInitializeSid(&SIDAuthNT, 2,
+        SECURITY_BUILTIN_DOMAIN_RID,
+        DOMAIN_ALIAS_RID_ADMINS,
+        0, 0, 0, 0, 0, 0,
+        &pAdminSID);
+    ok(bRes == TRUE, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (bRes == FALSE)
+        goto Cleanup;
+
+    // Initialize an EXPLICIT_ACCESS structure for an ACE.
+    // The ACE will allow the Administrators group full access to the key.
+    ea[1].grfAccessPermissions = KEY_ALL_ACCESS;
+    ea[1].grfAccessMode = SET_ACCESS;
+    ea[1].grfInheritance= NO_INHERITANCE;
+    ea[1].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[1].Trustee.TrusteeType = TRUSTEE_IS_GROUP;
+    ea[1].Trustee.ptstrName  = pAdminSID;
+
+    // Create a new ACL that contains the new ACEs.
+    dwRes = SetEntriesInAclW(_countof(ea), ea, NULL, &pACL);
+    ok(dwRes == ERROR_SUCCESS, "SetEntriesInAcl Error %ld\n", GetLastError());
+    if (dwRes != ERROR_SUCCESS)
+        goto Cleanup;
+
+    // Initialize a security descriptor.
+    pSD = LocalAlloc(LPTR, SECURITY_DESCRIPTOR_MIN_LENGTH);
+    ok(pSD != NULL, "LocalAlloc Error %ld\n", GetLastError());
+    if (pSD == NULL)
+        goto Cleanup;
+ 
+    bRes = InitializeSecurityDescriptor(pSD, SECURITY_DESCRIPTOR_REVISION);
+    ok(bRes == TRUE, "InitializeSecurityDescriptor Error %ld\n", GetLastError());
+    if (!bRes)
+        goto Cleanup;
+
+    // Add the ACL to the security descriptor.
+    bRes = SetSecurityDescriptorDacl(pSD,
+        TRUE,     // bDaclPresent flag
+        pACL,
+        FALSE);   // not a default DACL
+    ok(bRes == TRUE, "SetSecurityDescriptorDacl Error %ld\n", GetLastError());
+    if (!bRes)
+        goto Cleanup;
+
+    // Initialize a security attributes structure.
+    sa.lpSecurityDescriptor = pSD;
+    sa.bInheritHandle = FALSE;
+
+    // Use the security attributes to set the security descriptor
+    // when you create a key using sa.nLength of 0.
+    sa.nLength = 0;
+    lRes = RegCreateKeyExW(HKEY_CURRENT_USER, L"mykey", 0, L"", 0,
+            KEY_READ | KEY_WRITE, &sa, &hkSub, &dwDisposition);
+    ok(lRes == ERROR_SUCCESS, "RegCreateKeyExW returned '%ld', expected 0", lRes);
+
+    // Test the -A function
+    lRes = RegCreateKeyExA(HKEY_CURRENT_USER, "mykey", 0, "", 0,
+            KEY_READ | KEY_WRITE, &sa, &hkSub, &dwDisposition);
+    ok(lRes == ERROR_SUCCESS, "RegCreateKeyExA returned '%ld', expected 0", lRes);
+
+    // Use the security attributes to set the security descriptor
+    // when you create a key using sa.nLength of sa.nLength too short, but not 0.
```

Comment:
```suggestion
    // when you create a key using a too short sa.nLength, but not 0.
```

Reply:
Reworded for better readability. Thanks.